tests/store/api/conftest.py

```diff
@@ -180,6 +180,102 @@ def fixture_rml_store(store: Store, helpers: StoreHelpers) -> Store:
     return store
 
 
+@pytest.fixture(name="case_id_with_single_sample")
+def case_id_with_single_sample():
+    return "screamingtiger"
+
+
+@pytest.fixture(name="case_id_with_multiple_samples")
+def case_id_with_multiple_samples():
+    return "righteouspanda"
+
+
+@pytest.fixture(name="case_id_without_samples")
+def case_id_without_samples():
+    return "confusedtrout"
+
+
+@pytest.fixture(name="sample_id_in_single_case")
+def sample_id_in_single_case():
+    return "ASM1"
+
+
+@pytest.fixture(name="sample_id_in_multiple_cases")
+def sample_id_in_multiple_cases():
+    return "ASM2"
+
+
+@pytest.fixture(name="dummy_store")
+def fixture_dummy_store(
+    case_id_without_samples: str,
+    case_id_with_single_sample: str,
+    case_id_with_multiple_samples: str,
+    sample_id_in_single_case: str,
+    sample_id_in_multiple_cases: str,
+    helpers,
+    store: Store,
+):
+    sample_in_multiple_cases = helpers.add_sample(
+        store=store, internal_id=sample_id_in_multiple_cases
+    )
+    store.add_commit(sample_in_multiple_cases)
+
+    case_with_multiple_samples = helpers.add_case(
+        store=store, internal_id=case_id_with_multiple_samples, name="case_5"
+    )
+    store.add_commit(case_with_multiple_samples)
+
+    helpers.add_relationship(
+        store=store, sample=sample_in_multiple_cases, case=case_with_multiple_samples
+    )
+
+    seed_data = [
+        ("angrypanda", "AAM5", "case_1"),
+        ("jealousbutterfly", "AAM6", "case_2"),
+        ("lonesomespider", "AAM7", "case_3"),
+        ("happysnail", "AAM8", "case_4"),
+    ]
+
+    for data in seed_data:
+        case_id, sample_id, case_name = data
+
+        new_case = helpers.add_case(store=store, internal_id=case_id, name=case_name)
+        store.add_commit(new_case)
+
+        new_sample = helpers.add_sample(store=store, internal_id=sample_id)
+        store.add_commit(new_sample)
+
+        helpers.add_relationship(
+            store=store,
+            sample=new_sample,
+            case=new_case,
+        )
+
+        # Add association between new case and sample that should be present in multiple cases.
```

Comment:
nstead of comment could you define functions to loop and create these cases, sample and relations?

Reply:
Not sure what/how you mean!

Reply:
Should I extract a function ```add_case_with_single_sample``` and ```add_case_without_sample``` for instance?

Reply:
Simplified the mock using some helper functions.